test/functional/test-url.js

```diff
@@ -477,6 +478,25 @@ describe('isLocalhostOrigin', () => {
       'http://www.example.com/foo.html', false);
 });
 
+describe('isProtocolValid', () => {
+  function testProtocolValid(href, bool) {
+    it('should return whether it is a valid protocol for ' + href, () => {
+      expect(isProtocolValid(href)).to.equal(bool);
+    });
+  }
+
```

Comment:
test cases for:
-empty string
-non-Url string (e.g. `foo` ) (well technically this can be considered relative url)
-absolute but not fully qualified url   (`/foo/bar`)
-scheme-relative url (`//example.com/`)

all of these should be considered valid protocol

not sure what to do for `null`, `undefined`, `object`, etc.. I lean toward returning true instead of throwing, @jridgewell will have better ideas than me.